labelbox/schema/bulk_import_request.py

```diff
@@ -0,0 +1,335 @@
+        with file.open('rb') as f:
+            if validate_file:
+                data = f.read()
+                try:
+                    ndjson.loads(data)
+                except ValueError:
+                    raise ValueError(f"{file} is not a valid ndjson file")
+                file_data = (file.name, data, NDJSON_MIME_TYPE)
+            else:
+                file_data = (file.name, f, NDJSON_MIME_TYPE)
+            response_data = cls.__send_create_file_command(
+                client, request_data, file_name, file_data)
+        return cls.__build_bulk_import_request_from_result(
+            client, response_data["createBulkImportRequest"])
+
+    # TODO(gszpak): building query body should be handled by the client
+    @classmethod
+    def get(cls, client: Client, project_id: str, name: str) -> 'BulkImportRequest':
+        """
+        Fetches existing BulkImportRequest.
+
+        Args:
+            client (Client): a Labelbox client
+            project_id (str): BulkImportRequest's project id
+            name (str): name of BulkImportRequest
+        Returns:
+            BulkImportRequest object
+        """
+        query_str = """query getBulkImportRequestPyApi(
+                $projectId: ID!, $name: String!) {
+            bulkImportRequest(where: {
+                projectId: $projectId,
+                name: $name
+            }) {
+                %s
+            }
+        }
+        """ % cls.__build_results_query_part()
+        params = {
+            "projectId": project_id,
+            "name": name
+        }
+        bulk_import_request_kwargs = \
+            client.execute(query_str, params=params).get("bulkImportRequest")
+        if bulk_import_request_kwargs is None:
+            raise labelbox.exceptions.ResourceNotFoundError(
+                BulkImportRequest, {
+                    "projectId": project_id,
+                    "name": name
+                })
+        return cls.__build_bulk_import_request_from_result(client, bulk_import_request_kwargs)
+
+    def refresh(self) -> None:
+        """
+        Synchronizes values of all fields with the database.
+        """
+        bulk_import_request = self.get(self.client, self.project().uid, self.name)
+        for field in self.fields():
+            setattr(self, field.name, getattr(bulk_import_request, field.name))
+
+    def wait_until_done(
+            self, sleep_time_seconds: int = 30) -> None:
+        """
+        Blocks until the BulkImportRequest.state changes either to
+        `BulkImportRequestState.FINISHED` or `BulkImportRequestState.FAILED`,
+        periodically refreshing object's state.
+
+        Args:
+            sleep_time_seconds (str): a time to block between subsequent API calls
+        """
+        while self.state == BulkImportRequestState.RUNNING:
+            logger.info(f"Sleeping for {sleep_time_seconds}...")
+            self.__exponential_backoff_refresh()
+
+    @backoff.on_exception(
```

Comment:
did you find that `backoff` is better suited than google's `retry`?  https://googleapis.dev/python/google-api-core/latest/retry.html

Reply:
Worked pretty well - that was the first thing I saw when I googled "exponential backoff python", tried it with local API down and logs seemed reasonable